File: tests/fixtures/sample_repo/ruby_pkg/calculator.rb
# Calculator class for basic arithmetic
require_relative './models'

class Calculator < BaseCalculator
  def add(x, y)
    result = x + y
    record(x, y, result)
    result
  end

  def subtract(x, y)
    result = x - y
    record(x, y, result)
    result
  end

  private

  def record(x, y, result)
    @history << CalculationRecord.new(x, y, result)
  end
end

File: tests/fixtures/sample_repo/kotlin_pkg/Calculator.kt
package sample

import sample.models.Operation
import sample.models.CalculationRecord

class Calculator {
    private val history = mutableListOf<CalculationRecord>()

    fun add(x: Double, y: Double): Double {
        val result = x + y
        record(Operation.ADD, x, y, result)
        return result
    }

    fun subtract(x: Double, y: Double): Double {
        val result = x - y
        record(Operation.SUBTRACT, x, y, result)
        return result
    }

    private fun record(op: Operation, x: Double, y: Double, result: Double) {
        history.add(CalculationRecord(op, x, y, result))
    }

    fun getHistory(): List<CalculationRecord> = history
}
